projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-admin-menu.php

```diff
@@ -107,15 +107,31 @@ public function prepare_menu_for_response( array $menu ) {
 		 * See: https://github.com/Automattic/wp-calypso/blob/ebde236ec9b21ea9621c0b0523bd5ea185523731/client/state/admin-menu/schema.js
 		 */
 		foreach ( $menu as $menu_item ) {
-			$item = $this->prepare_menu_item( $menu_item );
+
+			/**
+			 * Logic extracted from _wp_menu_output() core function.
+			 * Based $admin_is_parent we determine if the submenu items should be pointing to the admin.php or not.
+			 *
+			 * @see https://core.trac.wordpress.org/browser/trunk/src/wp-admin/menu-header.php?rev=49193#L173
+			 */
+			$menu_hook = get_plugin_page_hook( $menu_item[2], 'admin.php' );
+			$menu_file = $menu_item[2];
+			$pos       = strpos( $menu_file, '?' );
+
+			if ( false !== $pos ) {
+				$menu_file = substr( $menu_file, 0, $pos );
+			}
+
+			$admin_is_parent = ! empty( $menu_hook ) || ( ( 'index.php' !== $menu_item[2] ) && file_exists( WP_PLUGIN_DIR . "/$menu_file" ) && ! file_exists( ABSPATH . "/wp-admin/$menu_file" ) );
+
+			$item = $this->prepare_menu_item( $menu_item, $admin_is_parent );
```

Comment:
Core does not check `$admin_is_parent` for parent menu items, so I think we shouldn't define it here.
```suggestion
			$item = $this->prepare_menu_item( $menu_item );
```